File: dummy_code/vulnerable_rsa.c
#include <openssl/rsa.h>
#include <openssl/pem.h>
#include <stdio.h>

int main() {
    RSA *key = RSA_generate_key(1024, RSA_F4, NULL, NULL); // 🔴 Weak RSA Key (1024-bit)

    FILE *priv_file = fopen("vulnerable_rsa_private.pem", "wb");
    PEM_write_RSAPrivateKey(priv_file, key, NULL, NULL, 0, NULL, NULL);
    fclose(priv_file);

    FILE *pub_file = fopen("vulnerable_rsa_public.pem", "wb");
    PEM_write_RSA_PUBKEY(pub_file, key);
    fclose(pub_file);

    RSA_free(key);
    printf("Generated Vulnerable RSA 1024-bit Key\\n");
    return 0;
}

File: dummy_code/vulnerable_rsa.py
from Crypto.PublicKey import RSA

# Generate a weak RSA key (1024-bit, vulnerable)
key = RSA.generate(1024)
private_key = key.export_key()
public_key = key.publickey().export_key()

print("Vulnerable RSA 1024-bit Private Key:")
print(private_key.decode())

print("Vulnerable RSA 1024-bit Public Key:")
print(public_key.decode())
